flink-metrics/flink-metrics-otel/src/test/java/org/apache/flink/metrics/otel/OpenTelemetryMetricReporterITCase.java

```diff
@@ -0,0 +1,276 @@
+                        Clock.fixed(Instant.ofEpochMilli(TIME_MS), Clock.systemUTC().getZone()));
+    }
+
+    @Test
+    public void testReport() throws Exception {
+        MetricConfig metricConfig = createMetricConfig();
+        MetricGroup group = new TestMetricGroup();
+
+        reporter.open(metricConfig);
+
+        SimpleCounter counter = new SimpleCounter();
+        reporter.notifyOfAddedMetric(counter, "foo.counter", group);
+
+        Gauge<Double> gauge = () -> 123.456d;
+        reporter.notifyOfAddedMetric(gauge, "foo.gauge", group);
+
+        reporter.report();
+
+        MeterView meter = new MeterView(counter);
+        reporter.notifyOfAddedMetric(meter, "foo.meter", group);
+
+        reporter.notifyOfAddedMetric(histogram, "foo.histogram", group);
+
+        reporter.report();
+        reporter.close();
+
+        eventuallyConsumeJson(
+                (json) -> {
+                    JsonNode scopeMetrics =
+                            json.findPath("resourceMetrics").findPath("scopeMetrics");
+                    assertThat(scopeMetrics.findPath("scope").findPath("name").asText())
+                            .isEqualTo("io.confluent.flink.common.metrics");
+                    JsonNode metrics = scopeMetrics.findPath("metrics");
+
+                    List<String> metricNames = extractMetricNames(json);
+                    assertThat(metricNames)
+                            .contains(
+                                    "flink.logical.scope.foo.counter",
+                                    "flink.logical.scope.foo.gauge",
+                                    "flink.logical.scope.foo.meter.count",
+                                    "flink.logical.scope.foo.meter.rate",
+                                    "flink.logical.scope.foo.histogram");
+
+                    metrics.forEach(OpenTelemetryMetricReporterITCase::assertMetrics);
+                });
+    }
+
+    private static void assertMetrics(JsonNode metric) {
+        String name = metric.findPath("name").asText();
+        if (name.equals("flink.logical.scope.foo.counter")) {
+            assertThat(metric.at("/sum/dataPoints").findPath("asInt").asInt()).isEqualTo(0);
+        } else if (name.equals("flink.logical.scope.foo.gauge")) {
+            assertThat(metric.at("/gauge/dataPoints").findPath("asDouble").asDouble())
+                    .isCloseTo(123.456, Percentage.withPercentage(1));
+        } else if (name.equals("flink.logical.scope.foo.meter.count")) {
+            assertThat(metric.at("/sum/dataPoints").findPath("asInt").asInt()).isEqualTo(0);
+        } else if (name.equals("flink.logical.scope.foo.meter.rate")) {
+            assertThat(metric.at("/gauge/dataPoints").findPath("asDouble").asDouble())
+                    .isEqualTo(0.0);
+        } else if (name.equals("flink.logical.scope.foo.histogram")) {
+            assertThat(metric.at("/summary/dataPoints").findPath("sum").asInt()).isEqualTo(4);
+        }
```

Comment:
Can't we have a mapping from metric name to the expected value instead?

Reply:
I'm slightly against that, as the assertions are a bit different  (`asInt`, `asDouble`, `sum`). It could be handled with lambda functions doing the whole thing like `findPath("asInt").asInt()).isEqualTo(0);`. This would be slightly nicer, but debugging and stack traces would be mangled. So if I had to add a couple more assertions, I would do it but with the current size I would keep it as is.